target_postgres/tests/test_target_postgres.py

```diff
@@ -45,7 +47,7 @@ def postgres_config_ssh_tunnel_fixture():
     return postgres_config_ssh_tunnel()
 
 
-@pytest.fixture
+@pytest.fixture(scope="session")
 def postgres_target(postgres_config) -> TargetPostgres:
     return TargetPostgres(config=postgres_config)
```

Comment:
Hi. We have been enjoying a conversation about a potential resource leak.

- GH-260

@BuzzCutNorman said:

> If I am following the test code correctly each test is creating an engine/connection pool and not closing or disposing of the engine. I think this is causing you to reach the limit of connections. My suggestion would be to dispose of the engine at the end of each test that generates an engine to see if that helps.

I just wanted to pick this up in order to outline that the problem has been solved differently here, by making the provided `TargetPostgres` object session-scoped. I think it is wrong, and hides any underlying issues. In that spirit, I think the fixes on behalf of GH-250 are more correct, as they are fixing details closer to the root cause by explicitly disposing SQLAlchemy engine objects.

Reply:
^^ This code has been removed again. The test fixtures are function-scoped again.